src/didhistory.c

```diff
@@ -0,0 +1,159 @@
+        DIDError_Set(DIDERR_INVALID_ARGS, "Invalid arguments.");
+        return NULL;
+    }
+
+    did = (DID*)calloc(1, sizeof(DID));
+    if (!did) {
+        DIDError_Set(DIDERR_OUT_OF_MEMORY, "Malloc buffer for DID failed.");
+        return NULL;
+    }
+
+    strcpy(did->idstring, history->did.idstring);
+    return did;
+}
+
+int DIDHistory_GetStatus(DIDHistory *history)
+{
+    if (!history) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Invalid arguments.");
+        return STATUS_NOT_FOUND;
+    }
+
+    return history->status;
+}
+
+ssize_t DIDHistory_GetTransactionCount(DIDHistory *history)
+{
+    if (!history) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Invalid arguments.");
+        return -1;
+    }
+
+    return history->txinfos.size;
+}
+
+DIDDocument *DIDHistory_GetTxDocumentByIndex(DIDHistory *history, int index)
+{
+    DIDDocument *doc;
+
+    if (!history || index < 0) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Invalid arguments.");
+        return NULL;
+    }
+
+    if (!history->txinfos.infos) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "No transaction in history.");
+        return NULL;
+    }
+
+    if (index >= history->txinfos.size) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Index is larger than transaction count in history.");
+        return NULL;
+    }
+
+    doc = (DIDDocument*)calloc(1, sizeof(DIDDocument));
+    if (!doc) {
+        DIDError_Set(DIDERR_OUT_OF_MEMORY, "Malloc buffer for document failed.");
+        return NULL;
+    }
+
+    if (DIDDocument_Copy(doc, history->txinfos.infos[index].request.doc) < 0) {
+        DIDDocument_Destroy(doc);
+        return NULL;
+    }
+
+    return doc;
+}
+
+const char *DIDHistory_GetTxIDByIndex(DIDHistory *history, int index)
+{
+    if (!history || index < 0) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Invalid arguments.");
+        return NULL;
+    }
+
+    if (!history->txinfos.infos || index >= history->txinfos.size) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Index is larger than transaction count in history.");
+        return NULL;
+    }
+
+    return strdup(history->txinfos.infos[index].txid);
+}
+
+time_t DIDHistory_GetTxPublishedByIndex(DIDHistory *history, int index)
+{
+    if (!history || index < 0) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Invalid arguments.");
+        return 0;
+    }
+
+    if (!history->txinfos.infos || index >= history->txinfos.size) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Index is larger than transaction count in history.");
+        return 0;
+    }
+
+    return history->txinfos.infos[index].timestamp;
+}
+
+const char *DIDHistory_GetTxOperationByIndex(DIDHistory *history, int index)
+{
+    if (!history || index < 0) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Invalid arguments.");
+        return NULL;
+    }
+
+    if (!history->txinfos.infos || index >= history->txinfos.size) {
+        DIDError_Set(DIDERR_INVALID_ARGS, "Index is larger than transaction count in history.");
```

Comment:
Should we have error number like :DIDERR_MALFORMED_HISTROY or just to use the error DIDERR_MALFORMED_RESOLVE_RESULT